android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java

```diff
@@ -447,7 +457,45 @@ private void isWiFiAPEnabled(Result poResult) {
      */
     private void setWiFiAPEnabled(MethodCall poCall, Result poResult) {
         boolean enabled = poCall.argument("state");
-        moWiFiAPManager.setWifiApEnabled(null, enabled);
+
+        /**
+         * Using LocalOnlyHotspotCallback when setting WiFi AP state on android Q and later
+         */
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q) {
+            moWiFiAPManager.setWifiApEnabled(null, enabled);
+        }
+            else {
+                if (enabled) {
+                    moWiFi.startLocalOnlyHotspot(new WifiManager.LocalOnlyHotspotCallback() {
+                        @Override
+                        public void onStarted(WifiManager.LocalOnlyHotspotReservation reservation) {
+                            super.onStarted(reservation);
+                            apReservation = reservation;
+                        }
+
+                        @Override
+                        public void onStopped() {
+                            super.onStopped();
+                            Log.d(WifiIotPlugin.class.getSimpleName(), "LocalHostpot Stopped.");
+                        }
+
+                        @Override
+                        public void onFailed(int reason) {
+                            super.onFailed(reason);
+                            Log.d(WifiIotPlugin.class.getSimpleName(), "LocalHotspot failed with code: " + String.valueOf(reason));
```

Comment:
do you think we should describe this behaviour in README?

Reply:
Though I can't still test this part of code, it says wifi AP doesn't support the device. I'll test this first.

Reply:
Done testing. It works.